### PR TITLE
Fixes #19471 - Update SSL path on puppet 4+

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

**Comment by xprazak2** [comment]
I am getting:
```
Error: Could not retrieve catalog from remote server: Error 400 on SERVER: Syntax error at 'else'; expected '}' at /etc/puppet/environments/production/modules/foreman_scap_client/manifests/init.pp:73 on node puppet-test.example.com
Warning: Not using cache on failed catalog
Error: Could not retrieve catalog; skipping run
```

when testing with 3.8.7

**Comment by ares** [comment]
it might be a different issue, would you mind trying deleting trailing comma on line 73 `        before => Yumrepo['foreman-plugins'],`?

**Comment by xprazak2** [comment]
Comma is not a problem, `else` with `unless` causes trouble. I found out this is not rebased on [Puppet 3 compatibility fix](https://github.com/theforeman/puppet-foreman_scap_client/commit/c986ee9adf2df35f9032e3f2df056659b4ae51c9). I will continue testing.

File: manifests/params.pp [hunk 1]
```diff
@@ -1,6 +1,12 @@
 class foreman_scap_client::params {
   $downcased_fqdn = downcase($fqdn)
-  $ca_file          = pick($::rh_certificate_repo_ca_file, '/var/lib/puppet/ssl/certs/ca.pem')
-  $host_certificate = pick($::rh_certificate_consumer_host_cert, "/var/lib/puppet/ssl/certs/${$downcased_fqdn}.pem")
-  $host_private_key = pick($::rh_certificate_consumer_host_key, "/var/lib/puppet/ssl/private_keys/${$downcased_fqdn}.pem")
+  if versioncmp('4.0.0', $clientversion) < 0 {
```
Comment (xprazak2) [inline manifests/params.pp]: I think here should be `>`

**Comment by ares** [comment]
[Of course](https://docs.puppet.com/puppet/latest/function.html#versioncmp), good catch. I rebased the PR and fix the comparison. @xprazak2 would you mind to take another look?

**Comment by xprazak2** [comment]
It works now as expected, merging. Thanks!

**Pull request merged**